.github/workflows/codeql.yml

```
@@ -0,0 +1,74 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  schedule:
+    - cron: '23 12 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: [ubuntu-latest, windows-latest, macos-latest]
```

This won't work properly. I believe we need to use a matrix.
```suggestion
    runs-on: ${{ matrix.os }}
```

---

> Are you sure?

I believe so. I've done it in other repositories and it doesn't work. When I use `matrix.os`, it works.

---

Oh I think you're right. I will see what I can do 😄